pilot/pkg/serviceregistry/external/servicediscovery.go

```diff
@@ -199,6 +199,25 @@ func getServiceEntryHandler(c *ServiceEntryStore) func(model.Config, model.Confi
 			}
 		}
 
+		if len(unchangedSvcs) > 0 {
+			// If this service entry had endpoints with IPs (i.e. resolution STATIC), then we do EDS update.
+			// If the service entry had endpoints with FQDNs (i.e. resolution DNS), then we need to do
+			// full push (as fqdn endpoints go via strict_dns clusters in cds).
+			currentServiceEntry := curr.Spec.(*networking.ServiceEntry)
+			oldServiceEntry := old.Spec.(*networking.ServiceEntry)
+			if currentServiceEntry.Resolution == networking.ServiceEntry_DNS {
+				if !reflect.DeepEqual(currentServiceEntry.Endpoints, oldServiceEntry.Endpoints) {
```

Comment:
what about hosts? does only endpoints matter?

Reply:
if hosts change, we catch it in the updatedServices section right?